### PR TITLE
Allow multi-line changelog entries

### LINKED ISSUE #38: Allow sub-bullets in change lists
Is there any reason not to allow this? There are no sub-bullets in the example in [the spec](https://keepachangelog.com/en/1.0.0/), but also nothing forbidding their existence.

```
### Changed

- #6: Bla bla bla
  - Details
```

Currently, `auto-changelog validate` will error on `  - Details`.

### PR DESCRIPTION
**Pull request by Gudahtt**
Changelog entries are now allowed to span multiple lines and contain sub-bullets. We even preserve trailing newlines in change descriptions, because why not? Enforcing newlines isn't required to make sense of the format.

Additional tests have been added to reflect the changes. A test has been added to reflect that newlines aren't required between sections in parsed changelogs either (this functionality was pre-existing, but the absence was highlighted by the coverage report).

Fixes #38

### CODE REVIEW

File: src/parseChangelog.js [hunk 1]
```diff
@@ -30,12 +30,44 @@ function parseChangelog({ changelogContent, repoUrl }) {
...
+   * Finalize a change entry, adding it to the changelog.
+   *
+   * This is required because change entries can span multiple lines.
+   *
+   * @param {Object} [options]
+   * @param {boolean} [options.removeTrailingNewline] - Indicates that the
+   *   trailing newline is not a part of the change description, so should be
+   *   removed.
+   */
+  function finalizePreviousChange({ removeTrailingNewline = false } = {}) {
```
Comment (shanejonas) [inline src/parseChangelog.js:52]: default values for destructuring is weird to read lol

File: src/parseChangelog.test.js [hunk 2]
```diff
@@ -196,6 +196,119 @@ describe('parseChangelog', () => {
...
+
+        The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+        and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+        ## [Unreleased]
+
+        ## [1.0.0] - 2020-01-01
+        ### Changed
+        - Something else
+        Further explanation of changes
```
Comment (shanejonas) [inline src/parseChangelog.test.js:213]: is a multiline of text valid?

ex:

```
        ## [1.0.0] - 2020-01-01
        ### Changed
        Hello World
        Further explanation of changes
```

Comment (Gudahtt) [reply src/parseChangelog.test.js:213]: Yep! Any number of lines will be accepted as a valid change description.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**